openrewrite/src/javascript/parser.ts

```diff
@@ -147,19 +149,18 @@ export class JavaScriptParser extends Parser {
                 continue;
             }
 
-            if (this.hasFlowAnnotation(sourceFile)) {
+            if (hasFlowAnnotation(sourceFile)) {
                 result.push(ParseError.build(this, input, relativeTo, ctx, new FlowSyntaxNotSupportedError(`Flow syntax not supported: ${input.path}`), null));
                 continue;
             }
 
-            // ToDo: uncomment code after tests fixing
-            // const syntaxErrors = this.checkSyntaxErrors(program, sourceFile);
-            // if (syntaxErrors.length > 0) {
-            //     syntaxErrors.forEach(
-            //         e => result.push(ParseError.build(this, input, relativeTo, ctx, new SyntaxError(`Compiler error:  ${e[0]} [${e[1]}]`), null))
-            //     );
-            //     continue;
-            // }
+            const syntaxErrors = checkSyntaxErrors(program, sourceFile);
+            if (syntaxErrors.length > 0) {
+                syntaxErrors.forEach(
+                    e => result.push(ParseError.build(this, input, relativeTo, ctx, new SyntaxError(`Compiler error:  ${e[0]} [${e[1]}]`), null))
```

Comment:
Could this end up creating multiple ParseErrors for one source file?

Reply:
Yes, there can be many ParseErrors for one source file